### ISSUE TITLE
Reducing Inconsistent Word Pronunciation Across Generations

### ISSUE THREAD

**Issue by kawshikbuet17** [k2-fsa/OmniVoice#237]
### Checks

- [x] This template is only for usage issues encountered.
- [x] I have thoroughly reviewed the project documentation but couldn't find information to solve my problem.
- [x] I have searched for existing issues, including closed ones, and couldn't find a solution.
- [x] I am using English to submit this issue to facilitate community communication.

### Environment Details

I have been using OmniVoice voice cloning as a Bengali TTS system for the past few weeks. After experimenting with inference steps, precision, and other generation parameters, I have been able to get fairly close to the desired output quality.

However, I am still facing one pronunciation consistency issue. Some words are pronounced correctly most of the time, but are occasionally mispronounced when generating the same or similar text.

For example, the Bengali word “বিকাশ” is pronounced approximately as “bee-kash”. The expected final sound is “sh”, as in the English word “ship”. However, in some generations, it is pronounced closer to “bee-kas”, where the final sound becomes “s”, as in “see”. Occasionally, the middle vowel is also shortened, making it sound closer to “bee-kosh.”

Since the model can already generate the correct pronunciation in many cases, it seems to be a consistency issue rather than an unsupported pronunciation.

I have tried increasing the inference steps, modifying the Bengali spelling, using Latin transliterations such as bikash, and experimenting with phoneme-based inputs. These approaches sometimes improve the result, but none of them completely eliminate the occasional error.

Could you please suggest the best approach to minimize this type of pronunciation inconsistency, especially for production use?

### Steps to Reproduce

N/A

### ❌ Actual Behavior

N/A

**Comment by zhu-han** [comment]
Hi. This model has randomness during generation. You can try fixing the random seed at inference. But it will only guarantee consistent outputs, not correct ones. This model may have occasional mispronunciations. For more accurate pronunciation, you can either fine-tune it on a high-quality dataset, or use phoneme-character mixed transcriptions for fine-tuning (as we did for English and Chinese) to fix frequently mispronounced words.

**Comment by kawshikbuet17** [comment]
@zhu-han 
Hi, can you please comment on my current hyperparameters? BTW, I'm using voice clone.
```
"gen_kwargs": {
        "speed": 1.0,
        "denoise": True,
        "num_step": 16,
        "guidance_scale": 2,
        "t_shift": 0.1,
        "preprocess_prompt": True,
        "postprocess_output": True,
        "layer_penalty_factor": 5.0,
        "position_temperature": 0.0,
        "class_temperature": 0.0,
        "audio_chunk_duration": 15.0,
        "audio_chunk_threshold": 20.0,
    }
```

If you have any suggestions regarding my hyperparameters, please share them.
Let me tell you one more thing. Sometimes I've noticed that the generated audio becomes slow/fast in a particular portion of a sentence, and the other portions' speed is fine. For example, a generated audio whose duration is around 8 seconds sometimes gets faster/slower in one portion, but the other portions are fine. This doesn't always happen. I have tried with good reference voices and reference texts, but sometimes the issue happens.

Thanks in advance.

**Comment by zhu-han** [comment]
Most of your hyperparameters match the defaults. One risky configuration is position_temperature=0, as it may cause poor speech quality for some challenging samples. Besides, num_step=16 trades quality for speed. I recommend keeping all hyperparameters as defaults if you want the best performance.

**Comment by kawshikbuet17** [comment]
@zhu-han thanks. One last thing. Suppose I want to tune the hyperparameters to reduce the inference time; a little reduction in clone quality is accepted, but pronunciation accuracy is a must. Then, which hyperparameter(s) should I mess with?

**Comment by zhu-han** [comment]
`num_step` is related to inference time. Fewer steps yield faster inference but may result in lower quality.
